configs/next/packages/libnxz/stage_1

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# Copyright 2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+source ${AT_BASE}/scripts/utilities/bitsize_selection.sh
+
+# NX GZIP library build parameters for stage 1  64 bits
+# ======================================================================
+#
+
+# Tell the build system to hold the temp install folder
+ATCFG_HOLD_TEMP_INSTALL='no'
+# Tell the build system to hold the temp build folder
+ATCFG_HOLD_TEMP_BUILD='no'
+# Build in a new directory
+ATCFG_BUILD_STAGE_T='dir'
+
+atcfg_pre_configure() {
+	# The bin and lib directories depend on the word size, e.g.
+	# configure_bindir may be set to bin, bin32, or bin64.
+	configure_libdir=${configure_prefix}/$(find_build_libdir ${AT_BIT_SIZE})
+}
+
+atcfg_configure() {
+	# libnxz do not use autoconf
+	${SUB_MAKE} clean -C ${ATSRC_PACKAGE_WORK}
+	true
+}
+
+atcfg_make() {
+	# In gcc10 the default is now -fno-common. In order to not have multiple
+	# definitions -fcommon is needed.
+	configure_cc="${at_dest}/bin/${target64:-${target}}-gcc"
+	PATH="${at_dest}/bin:${PATH}" \
+	${SUB_MAKE} -C ${ATSRC_PACKAGE_WORK} \
+	CC="${configure_cc} -m${AT_BIT_SIZE}" \
+	FLG="-std=gnu11 -fcommon" ZLIB=""
+}
+
+atcfg_make_check() {
+	# Package testing is not done on a cross build.
+	if [[ "${cross_build}" == 'no' ]]; then
+		PATH=${at_dest}/bin:${PATH} ${SUB_MAKE} check -C ${ATSRC_PACKAGE_WORK}
+	fi
+}
+
+atcfg_pre_install() {
+	mkdir -p ${install_transfer}/${configure_libdir}
+	mkdir -p ${install_transfer}/include/libnxz
+}
+
+atcfg_install() {
+	# libnxz does not have a make install
+	install -m 755 ${ATSRC_PACKAGE_WORK}/libnxz.so ${install_transfer}/${configure_libdir}/libnxz.so
+	install -m 644 ${ATSRC_PACKAGE_WORK}/libnxz.a ${install_transfer}/${configure_libdir}/libnxz.a
+	install -m 644 ${ATSRC_PACKAGE_WORK}/inc_nx/* ${install_transfer}/include/libnxz/
```

Comment:
I don't think this line is correct.
@rzinsly Any idea which headers should be installed?

Reply:
nx-zlib.h, note that it will be under lib/ in next releases
Also the ones in inc_nx if you want to be able to run the codes at samples/, note that some use internal symbols. At least if we want the internal debug log routines inc_nx/nx_dbg.h.

Reply:
I think everything is covered.

Reply:
> Also the ones in inc_nx if you want to be able to run the codes at samples/

Sounds like these headers should not be installed.

Reply:
@tuliom, @rzinsly The codes at sample/ aren't examples usage? 
If so, either the includes are essential.

Reply:
@lucmaga They are indeed examples. IMHO, they are **not** state of the art examples, though.
They're using non-exported functions and internal structures that should not be used in examples.
IMHO, real user code should not be written that way.
With that said, I do believe I could be missing some functions or structures that should indeed be made public.  However, it's easier to not export them now (i.e. without installing headers) and start export them after we confirm the mistake.  It's more complicated to stop exporting an internal structure or function after users started depending on them.

Reply:
> However, it's easier to not export them now (i.e. without installing headers) and start export them after we confirm the mistake. It's more complicated to stop exporting an internal structure or function after users started depending on them.

Agreed.